aicsimageio/tests/readers/test_default_reader.py

```diff
@@ -1,37 +1,110 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
+import numpy as np
 import pytest
+from dask.diagnostics import Profiler
+from psutil import Process
 
 from aicsimageio import exceptions
 from aicsimageio.readers.default_reader import DefaultReader
 
 
-@pytest.mark.parametrize("filename", [
-    "example.bmp",
-    "example.png",
-    "example.jpg",
-    "example.gif"
+@pytest.mark.parametrize("filename, expected_shape, expected_dims, expected_chunksize, expected_task_count", [
+    ("example.bmp", (480, 640, 4), "YXC", (480, 640, 4), 0),
+    ("example.png", (800, 537, 4), "YXC", (800, 537, 4), 0),
+    ("example.jpg", (452, 400, 3), "YXC", (452, 400, 3), 0),
+    # Task count for multiple image formats should be 2 * number of images in file
+    ("example.gif", (72, 268, 268, 4), "TYXC", (1, 268, 268, 4), 144),
+    pytest.param(
+        "example.txt",
+        None,
+        None,
+        None,
+        None,
+        marks=pytest.mark.raises(exception=exceptions.UnsupportedFileFormatError)
+    )
 ])
-def test_default_reader_get_default_dims(resources_dir, filename):
+def test_default_reader(
+    resources_dir,
+    filename,
+    expected_shape,
+    expected_dims,
+    expected_chunksize,
+    expected_task_count
+):
     # Get file
     f = resources_dir / filename
 
-    # Open
-    with DefaultReader(f) as r:
-        # Dims should be set to 3D for all of these images
-        assert r.dims == "YXC"
-        assert r.metadata is None
-        assert DefaultReader.is_this_type(f)
+    # Read file
+    img = DefaultReader(f)
 
+    # Check that there are no open file pointers after init
+    proc = Process()
+    assert str(f) not in [f.path for f in proc.open_files()]
 
-@pytest.mark.parametrize("expected", [
-    "XYC",
-    "STC",
-    pytest.param("HELLOWORLD", marks=pytest.mark.raises(exception=exceptions.InvalidDimensionOrderingError)),
-    pytest.param("NO", marks=pytest.mark.raises(exception=exceptions.InvalidDimensionOrderingError))
+    # Check basics
+    with Profiler() as prof:
+        assert img.dims == expected_dims
+        assert img.metadata
+        assert img.dask_data.shape == expected_shape
+        assert img.dask_data.chunksize == expected_chunksize
+        # Check that basic details don't require task computation
+        assert len(prof.results) == 0
+
+    # Check that there are no open file pointers after basics
+    assert str(f) not in [f.path for f in proc.open_files()]
+
+    # Check computed type is numpy array, computed shape is expected shape, and task count is expected
+    with Profiler() as prof:
+        assert isinstance(img.data, np.ndarray)
+        assert img.data.shape == expected_shape
+        assert len(prof.results) == expected_task_count
```

Comment:
is expected_task_count a dask implementation detail?

Reply:
Yep it is how we know the task graph for actually retrieving and interacting with the array is optimized.